include/aos/common/crypto.hpp

```diff
@@ -247,6 +262,717 @@ class ProviderItf {
 };
 
 } // namespace x509
+
+namespace elliptic {
+/**
+ * A Curve represents a short-form Weierstrass curve.
+ */
+struct Curve { };
+
+/**
+ * P384 returns a Curve which implements NIST P-384 (FIPS 186-3, section D.2.4), also known as secp384r1.
+ */
+inline Curve P384()
+{
+    return {};
+}
+} // namespace elliptic
+
+namespace pkcs11 {
+
+/**
+ * Maximum length of PKCS11 slot description.
+ */
+constexpr auto cSlotDescriptionLen = AOS_CONFIG_CRYPTO_PKCS11_SLOT_DESCRIPTION_LEN;
+
+/**
+ * Maximum length of PKCS11 manufacture ID.
+ */
+constexpr auto cManufacturerIDLen = AOS_CONFIG_CRYPTO_PKCS11_MANUFACTURE_ID_LEN;
+
+/**
+ * Maximum length of PKCS11 token label.
+ */
+constexpr auto cTokenLabelLen = AOS_CONFIG_CRYPTO_PKCS11_TOKEN_LABEL_LEN;
+
+/**
+ * Library description length.
+ */
+constexpr auto cLibraryDescLen = AOS_CONFIG_CRYPTO_PKCS11_LIBRARY_DESC_LEN;
+
+/**
+ * HSM device model length.
+ */
+constexpr auto cModelLen = AOS_CONFIG_CRYPTO_PKCS11_MODEL_LEN;
+
+/**
+ * Length of HSM device serial number.
+ */
+constexpr auto cDeviceSerialLen = AOS_CONFIG_CRYPTO_PKCS11_DEVICE_SERIAL_LEN;
+
+/**
+ * Maximum length of user PIN (password).
+ */
+static constexpr auto cPINLength = AOS_CONFIG_CRYPTO_PKCS11_PIN_LEN;
+
+/**
+ * Generates random unique PIN.
+ *
+ * @param[out] pin result pin.
+ * @return Error.
+ */
+inline Error GenPIN(String& pin)
```

Comment:
This function could be useful not only in pkcs11 context. We have to move it to som common part.